### PR TITLE
Improve/add use_item documentation

### PR DESCRIPTION
**Pull request by DJMcNab**
Adds some documentation to use_item explaining all code paths (use imports are hard, especially with the ongoing discussion of anchored v. uniform paths - see https://github.com/rust-lang/rust/issues/55618 for what appears to be the latest developments)

### CODE REVIEW

File: crates/ra_syntax/src/grammar/items/use_item.rs [hunk 1]
```diff
@@ -7,25 +7,75 @@ pub(super) fn use_item(p: &mut Parser) {
...
+        // use *;
+        // use ::*;
+        // use some::path::{*};
+        // use some::path::{::*};
         (STAR, _) => p.bump(),
         (COLONCOLON, STAR) => {
+            // Parse `use ::*;`, which imports all from the crate root in Rust 2015
+            // This is invalid inside a use_tree_list, (e.g. `use some::path::{::*}`)
+            // but still parses and errors later: ('crate root in paths can only be used in start position')
+            // TODO: Add this error (if not out of scope)
```
Comment (matklad) [inline crates/ra_syntax/src/grammar/items/use_item.rs]: We should add this error checking, but to the validator and not the parser:

https://github.com/rust-analyzer/rust-analyzer/blob/0b1c0ee225a075d25d700fcd1c007615556e2afb/crates/ra_syntax/src/validation/mod.rs

**Comment by matklad** [comment]
Excellent, thanks!

bors r+

File: crates/ra_syntax/src/grammar/items/use_item.rs (hunk 1 above)
Comment (DJMcNab) [reply crates/ra_syntax/src/grammar/items/use_item.rs]: Yeah, it was just an overall comment that it was something that naturally comes up there, but should be added somewhere else.

**Pull request merged**